packages/yew/src/functional/hooks/use_callback.rs

```diff
@@ -68,8 +72,18 @@ pub fn use_callback<IN, OUT, F, D>(f: F, deps: D) -> Callback<IN, OUT>
 where
     IN: 'static,
     OUT: 'static,
-    F: Fn(IN) -> OUT + 'static,
+    F: Fn(IN, &D) -> OUT + 'static,
     D: PartialEq + 'static,
 {
-    (*use_memo(move |_| Callback::from(f), deps)).clone()
+    let deps = Rc::new(deps);
```

Comment:
Maybe we can use `D: PartialEq + 'static + Clone` to avoid this allocation, since in most cases the deps already implements `Clone`.

Reply:
I think Rc Clone is usually more cheap than Clone of deps, as Rc is just like a pointer, and we don't want to constrain Clone to end user. and let's be consistent with use_state_eq/use_effect_with_deps etc.

Reply:
1. Not really, since this hook is used in the case that deps is not changed frequently, assume the ratio is 100:1. when this hook is called 100 times, using Rc needs 100 alloc + 100 dealloc, but using Clone just needs 1 `deps.clone` since the Callback is memoized. 
2. For consistence, the `use_context` hook also uses `Clone`， so this is not a special one.
3. All the `deps` params of `use_effect_with_deps` in this repo implement Clone, so users are not constrained actually.
4. If in some rare cases users provide a not Clone deps, they can call Rc::new at the calling site.